src/xcode/ENA/ENA/Source/Client/HTTP Client/CachingHTTPClient.swift

```diff
@@ -141,6 +141,36 @@ class CachingHTTPClient: AppConfigurationFetching, StatisticsFetching, QRCodePos
 		}
 	}
 	
+	// MARK: VaccinationValueSetsFetching
+	
+	func fetchVaccinationValueSets(
+		etag: String?,
+		completion: @escaping VaccinationValueSetsCompletionHandler
+	) {
+		// Manual ETagging because we don't use native cache
+		var headers: [String: String]?
+		if let etag = etag {
+			headers = ["If-None-Match": etag]
+		}
+
+		session.GET(configuration.VaccinationValueSets, extraHeaders: headers) { result in
+			switch result {
+			case .success(let response):
+				do {
+					let package = try self.verifyPackage(in: response)
+					let vaccinationValueSetsData = try SAP_Internal_Dgc_ValueSets(serializedData: package.bin)
+					let responseETag = response.httpResponse.value(forCaseInsensitiveHeaderField: "ETag")
+					let vaccinationValueSetsResponse = VaccinationValueSetsResponse(vaccinationValueSetsData, responseETag)
+					Log.info("Received value sets: \(try vaccinationValueSetsData.jsonString())", log: .vaccination)
+					completion(.success(vaccinationValueSetsResponse))
+				} catch {
+					completion(.failure(error))
```

Comment:
add a log with the error?

Reply:
I added it in the completion block in VaccinationValueSetsProvider.swift

Reply:
okay. But if some other class is using that GET, the dev has to be clear that he should not forget the logs. So in my opinion, the logs should be at minimum in the GET call and if you want also in the completion block where it is used. But you can decide how to handle it.